db/compaction_job.cc

```diff
@@ -171,12 +171,18 @@ struct CompactionJob::SubcompactionState {
     const std::vector<FileMetaData*>& grandparents = compaction->grandparents();
 
     // Scan to find earliest grandparent file that contains key.
+    int grandparent_index_moved = 0;
     while (grandparent_index < grandparents.size() &&
            icmp->Compare(internal_key,
                          grandparents[grandparent_index]->largest.Encode()) >
                0) {
       if (seen_key) {
         overlapped_bytes += grandparents[grandparent_index]->fd.GetFileSize();
+        if (grandparent_index == 0 ||
+            grandparents[grandparent_index]->fd.GetFileSize() >
+                compaction->max_output_file_size() / 8) {
```

Comment:
Maybe I am missing something, but should not this condition
`grandparents[grandparent_index]->fd.GetFileSize() > compaction->max_output_file_size() / 8`
be true most of the time ?

Does that mean that we limit the number of grandparents to be maximum 2 files ?